src/FwdState.cc

```diff
@@ -240,11 +240,7 @@ FwdState::updateAleWithFinalError()
     if (!err || !al)
         return;
 
-    LogTagsErrors lte;
-    if (err->xerrno == ETIMEDOUT || err->type == ERR_READ_TIMEOUT)
-        lte.timedout = true;
-    else if (err->type != ERR_NONE)
-        lte.aborted = true;
+    const auto lte = LogTagsErrors::FromErrno(err->type == ERR_READ_TIMEOUT ? ETIMEDOUT : err->xerrno);
```

Comment:
Eduard and I tried to figure out why this code treated ERR_NONE `err->type` specially (commit 21f90159029aeb05fca55a6f48f120afb5f2b705), but failed. I speculate that this code was negatively influenced by Ftp::Client::failed() or similar ugly methods that try hard to trick the reader into thinking that the method creates ErrorState objects with ERR_NONE `type`.

We believe it is best to move on to simpler and less duplicated code here until we find specific reasons to make exceptions (which we will have to explain when re-adding).

This comment does not request any code changes. I am just explaining why PR code ignores the special ERR_NONE condition in this refactoring.

Reply:
I think it was there because client disconnect after sending the whole reply is not an abort/error.
Though I am also unable to easily find any path which leads there without setting the `err->type`.

```
FwdState::completed()
{
...
    if (entry->store_status == STORE_PENDING) {
        if (entry->isEmpty()) {
...
        } else {
            updateAleWithFinalError(); // if any
            if (storedWholeReply_)
                entry->completeSuccessfully(storedWholeReply_);
...
```